File: app/lib/domain/usecases/user/get_current_user_usecase.dart
import 'package:app/domain/core/failures.dart';
import 'package:app/domain/entities/user/user.dart';
import 'package:app/domain/repositories/user/user_repository.dart';
import 'package:dartz/dartz.dart';

class GetCurrentUserUseCase {
  final UserRepository _userRepository;
  GetCurrentUserUseCase(this._userRepository);
  Future<Either<Failure, User>> call() async {
    return _userRepository.getCurrentUser();
  }
}

File: frontend/ReactNative/src/domain/usecases/GetCurrentUserUseCase.ts
import { User } from '../entities/User';
import { UserRepository } from '../repositories/UserRepository';

export class GetCurrentUserUseCase {
  constructor(private userRepository: UserRepository) {}

  async execute(): Promise<User | null> {
    try {
      return await this.userRepository.getCurrentUser();
    } catch (error) {
      console.error('GetCurrentUserUseCase error:', error);
      throw new Error('Failed to get current user');
    }
  }
} 